src/CommunityPatch/Patches/AgentWeaponEquippedPatch.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.DotNet;
+using TaleWorlds.Engine;
+using TaleWorlds.MountAndBlade;
+
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches {
+
+  public abstract class AgentWeaponEquippedPatch : PatchBase<AgentWeaponEquippedPatch> {
+
+    private static readonly MethodInfo TargetMethodInfo = typeof(Agent).GetMethod("WeaponEquipped", BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+
+    private static readonly MethodInfo PatchMethodInfo = typeof(AgentWeaponEquippedPatch).GetMethod(nameof(PostFix), BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+
+    public override bool Applied { get; protected set; }
+
+    public override void Apply(Game game) {
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    public override bool IsApplicable(Game game) {
+      if (Applied) return false;
+
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo))
+        return false;
+
+      return _IsApplicable(game);
+    }
+
+    private static void PostFix(ref Agent __instance,
```

Comment:
Given the Patch call above now applies this as the prefix parameter, does this need to be renamed to Prefix?

Also, postfix (proper; suffix) and prefix are full words, so Pascal naming convention would be to just capitalize the P.

Reply:
Fixed